foreman/data_refinery_foreman/foreman/main.py

```diff
@@ -181,17 +181,25 @@ def retry_lost_downloader_jobs() -> None:
     lost_jobs = []
     for job in potentially_lost_jobs:
         try:
-            job_status = nomad_client.job.get_job(job.nomad_job_id)["Status"]
-            # If the job is still pending, then it makes sense that it
-            # hasn't started and if it's running then it may not have
-            # been able to mark the job record as started yet.
-            if job_status != "pending" and job_status != "running":
-                logger.info(("Determined that a downloader job needs to be requeued because its"
-                             " Nomad Job's status is: %s."),
-                            job_status,
-                            job_id=job.id
-                )
-                lost_jobs.append(job)
+            if job.nomad_job_id:
+                job_status = nomad_client.job.get_job(job.nomad_job_id)["Status"]
+                # If the job is still pending, then it makes sense that it
+                # hasn't started and if it's running then it may not have
+                # been able to mark the job record as started yet.
+                if job_status != "pending" and job_status != "running":
+                    logger.info(("Determined that a downloader job needs to be requeued because its"
+                                 " Nomad Job's status is: %s."),
+                                job_status,
+                                job_id=job.id
+                    )
+                    lost_jobs.append(job)
+            else:
+                # If there is no nomad_job_id field set, we could be
+                # in the small window where the job was created but
+                # hasn't yet gotten a chance to be queued.
+                # If this job really should be restarted we'll get it in the next loop.
+                if timezone.now() - job.created_at > MIN_LOOP_TIME:
```

Comment:
This needs a test. I don't think these datetimes are all in the same format, so I think this might not do quite what you want. MIN_LOOP_TIME likely needs conversion to a timedelta